Source/FieldSolver/SpectralSolver/SpectralAlgorithms/PsatdAlgorithm.cpp

```diff
@@ -297,4 +296,79 @@ PsatdAlgorithm::CurrentCorrection( SpectralFieldData& field_data,
     field_data.BackwardTransform( *current[1], Idx::Jy, 0 );
     field_data.BackwardTransform( *current[2], Idx::Jz, 0 );
 }
+
+void
+PsatdAlgorithm::VayDeposition (SpectralFieldData& field_data,
+                               std::array<std::unique_ptr<amrex::MultiFab>,3>& current) {
+    // Profiling
+    WARPX_PROFILE("PsatdAlgorithm::VayDeposition");
+
+    using Idx = SpectralFieldIndex;
+
+    // Forward Fourier transform of D (temporarily stored in current):
+    // D is nodal and does not match the staggering of J
```

Comment:
Could you add maybe one more line of comment
```suggestion
    // D is nodal and does not match the staggering of J
    // Therefore, we pass the actual staggering of D (IntVect(1)) to the ForwardTransform function
```